src/Microsoft.Health.Fhir.Core/Features/Search/Expressions/Parsers/ExpressionParser.cs

```diff
@@ -101,7 +102,14 @@ public IncludeExpression ParseInclude(string resourceType, string includeValue,
                 refSearchParameter = _searchParameterDefinitionManager.GetSearchParameter(originalType.ToString(), searchParam.ToString());
             }
 
-            return new IncludeExpression(resourceType, refSearchParameter, originalType.ToString(), targetType, wildCard, isReversed);
+            referencedTypes = new List<string>();
+            _searchParameterDefinitionManager.GetSearchParameters(resourceType)
+            .Where(p => p.Type == ValueSets.SearchParamType.Reference).ToList()
+            .ForEach(p => referencedTypes.AddRange(p.TargetResourceTypes?.ToList()));
+
+            referencedTypes = referencedTypes.Distinct().ToList();
```

Comment:
what about this, should this be done only if iterate == true?

Reply:
De facto, it is used for expressions that are wildcards. And then :iterate expressions that come after can know what are the possible ctes to select from.